src/plugins/discover/public/utils/get_fields_to_show.ts

```diff
@@ -9,23 +9,30 @@ import { getFieldSubtypeMulti } from '@kbn/data-views-plugin/public';
 import type { DataView } from '@kbn/data-views-plugin/public';
 
 export const getFieldsToShow = (fields: string[], dataView: DataView, showMultiFields: boolean) => {
+  if (showMultiFields) {
+    return fields;
+  }
+  const fieldMap = {} as Record<string, boolean>;
   const childParentFieldsMap = {} as Record<string, string>;
+  const parentFieldMap = {} as Record<string, boolean>;
   const mapping = (name: string) => dataView.fields.getByName(name);
   fields.forEach((key) => {
     const mapped = mapping(key);
     const subTypeMulti = mapped && getFieldSubtypeMulti(mapped.spec);
+    const isMultiField = Boolean(subTypeMulti?.multi);
     if (mapped && subTypeMulti?.multi?.parent) {
-      childParentFieldsMap[mapped.name] = subTypeMulti.multi.parent;
+      childParentFieldsMap[key] = subTypeMulti.multi.parent;
```

Comment:
I don't think these two are necessarily the same.

Reply:
we're getting the mapped already by `key`, and the `mapping` function uses `getByName` for get the field, so I assume map === key in this case?